oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/name/ReadWriteNamespaceRegistry.java

```diff
@@ -38,8 +39,17 @@ public abstract class ReadWriteNamespaceRegistry
 
     private static final Logger LOG = LoggerFactory.getLogger(ReadWriteNamespaceRegistry.class);
 
-    public ReadWriteNamespaceRegistry(Root root) {
+    /**
+     * Feature flag to allow registering invalid namespace URIs (without a colon).
+     * Set the system property {@code oak.allowInvalidNamespaceUris} to {@code true} to enable this feature.
+     */
+    private final boolean allowInvalidNamespaceUris;
+
+    protected ReadWriteNamespaceRegistry(Root root) {
         super(root);
+        // cannot be static in order to allow testing with different values
+        allowInvalidNamespaceUris = SystemPropertySupplier.create("oak.allowInvalidNamespaceUris", false)
```

Comment:
can we move that up to the declaration of the variable?

Reply:
Done in https://github.com/apache/jackrabbit-oak/pull/2258/commits/b83396b5a8ba371745a7a55cdf30e0cd279e1fda.